ballerina/llm_client_azure_open_ai.bal

```diff
@@ -24,41 +25,90 @@ public type AzureOpenAIModelConfig record {|
+    string description?;
+    string name;
+    json schema?;
+    boolean? strict = true;
+|};
+
+type ChatCompletionAzureResponse record {
+    AzureOpenAIResponseMessage[] choices?;
+};
+
+type AzureOpenAIResponseMessage record {
+    record {
+        string? content;
+    } message;
+};
+
 # Azure OpenAI model chat completion client.
 public isolated distinct client class AzureOpenAIModel {
     *Model;
 
-    private final chat:Client cl;
     private final string deploymentId;
     private final string apiVersion;
+    private final string apiKey;
+    private final http:Client azureOpenAIClient;
 
-    public isolated function init(chat:Client|AzureOpenAIModelConfig azureOpenAI,
+    public isolated function init(AzureOpenAIModelConfig azureOpenAI,
             string deploymentId,
             string apiVersion) returns error? {
-        self.cl = azureOpenAI is chat:Client ?
-            azureOpenAI :
-            check new (azureOpenAI.connectionConfig, azureOpenAI.serviceUrl);
         self.deploymentId = deploymentId;
         self.apiVersion = apiVersion;
+        http:BearerTokenConfig|chat:ApiKeysConfig auth = azureOpenAI.connectionConfig.auth;
+
+        if auth is http:BearerTokenConfig {
+            self.apiKey = auth.token;
+        } else {
+            self.apiKey = auth.apiKey;
+        }
+
+        self.azureOpenAIClient = check generateHttpClientFromAzureOpenAIModelConfig(azureOpenAI);
     }
 
     isolated remote function call(string prompt, map<json> expectedResponseSchema) returns json|error {
-        chat:CreateChatCompletionRequest chatBody = {
-            messages: [{role: "user", "content": getPromptWithExpectedResponseSchema(prompt, expectedResponseSchema)}]
+        ChatCompletionAzureRequest chatBody = {
+            messages: [{role: "user", "content": getPromptWithExpectedResponseSchema(prompt, expectedResponseSchema)}],
+            response_format: check getJsonSchemaResponseFormatForAzureOpenAI(expectedResponseSchema)
         };
 
-        chat:Client cl = self.cl;
-        chat:CreateChatCompletionResponse chatResult =
-            check cl->/deployments/[self.deploymentId]/chat/completions.post(self.apiVersion, chatBody);
-        record {
-            chat:ChatCompletionResponseMessage message?;
-            chat:ContentFilterChoiceResults content_filter_results?;
-            int index?;
-            string finish_reason?;
-        }[]? choices = chatResult.choices;
+        string resourcePath = string
+            `/deployments/${getEncodedUri(self.deploymentId)}/chat/completions?api-version=${self.apiVersion}`;
+        http:Request request = new;
+        request.setPayload(chatBody);
+
+        ChatCompletionAzureResponse|error chatResult = self.azureOpenAIClient->post(
+                                            resourcePath, request, {api\-key: self.apiKey});
+
+        if chatResult is error {
+            return error("Chat completion failed");
```

Comment:
Include at least the error message from the original error.

Reply:
Added as an detail